src/components/OptionsBar/Options/FilterModal/FilterModalPage/FilterModalHeader.jsx

```diff
@@ -0,0 +1,56 @@
+import PropTypes from 'prop-types';
+import React, { useEffect, useRef } from 'react';
+import { Modal, Tab, Tabs } from 'react-bootstrap';
+import InputFilters from '../../../../InputFilters/InputFilters';
+
+function FilterModalHeader({ labels, index, setIndex, setFilteredText }) {
+  const filteredTextTabs = useRef([]);
+
+  const updateRelativeFilteredText = (text) => {
+    filteredTextTabs[index] = text;
+    setFilteredText(text);
+  };
+
+  useEffect(() => {
+    if (filteredTextTabs[index]) {
+      setFilteredText(filteredTextTabs[index]);
+    } else {
+      setFilteredText('');
+    }
```

Comment:
You can directly call the function with the value, and declare a default

```suggestion
    setFilteredText(filteredTextTabs[index] ?? '')
```